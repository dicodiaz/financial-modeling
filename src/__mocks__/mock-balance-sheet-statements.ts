import { BalanceSheetStatementType } from '../redux/utils/types';

const mockBalanceSheetStatement: BalanceSheetStatementType[] = [
  {
    date: '2022-10-31',
    symbol: 'A',
    reportedCurrency: 'USD',
    cik: '0001090872',
    fillingDate: '2022-12-21',
    acceptedDate: '2022-12-20 18:42:30',
    calendarYear: '2022',
    period: 'FY',
    cashAndCashEquivalents: 1053000000,
    shortTermInvestments: 0,
    cashAndShortTermInvestments: 1053000000,
    netReceivables: 1405000000,
    inventory: 1038000000,
    otherCurrentAssets: 282000000,
    totalCurrentAssets: 3778000000,
    propertyPlantEquipmentNet: 1100000000,
    goodwill: 4773000000,
    intangibleAssets: 821000000,
    goodwillAndIntangibleAssets: 4773000000,
    longTermInvestments: 195000000,
    taxAssets: 0,
    otherNonCurrentAssets: 670000000,
    totalNonCurrentAssets: 6738000000,
    otherAssets: 0,
    totalAssets: 10516000000,
    accountPayables: 580000000,
    shortTermDebt: 36000000,
    taxPayables: 0,
    deferredRevenue: 461000000,
    otherCurrentLiabilities: 784000000,
    totalCurrentLiabilities: 1861000000,
    longTermDebt: 2733000000,
    deferredRevenueNonCurrent: 0,
    deferredTaxLiabilitiesNonCurrent: 0,
    otherNonCurrentLiabilities: 633000000,
    totalNonCurrentLiabilities: 3366000000,
    otherLiabilities: 0,
    capitalLeaseObligations: 0,
    totalLiabilities: 5227000000,
    preferredStock: 0,
    commonStock: 3000000,
    retainedEarnings: 324000000,
    accumulatedOtherComprehensiveIncomeLoss: -363000000,
    othertotalStockholdersEquity: 5325000000,
    totalStockholdersEquity: 5289000000,
    totalEquity: 5289000000,
    totalLiabilitiesAndStockholdersEquity: 10516000000,
    minorityInterest: 0,
    totalLiabilitiesAndTotalEquity: 10516000000,
    totalInvestments: 195000000,
    totalDebt: 2769000000,
    netDebt: 1716000000,
    link: 'https://www.sec.gov/Archives/edgar/data/1090872/000109087222000026/0001090872-22-000026-index.htm',
    finalLink: 'https://www.sec.gov/Archives/edgar/data/1090872/000109087222000026/a-20221031.htm',
  },
  {
    date: '2021-10-31',
    symbol: 'A',
    reportedCurrency: 'USD',
    cik: '0001090872',
    fillingDate: '2021-12-17',
    acceptedDate: '2021-12-17 17:01:49',
    calendarYear: '2021',
    period: 'FY',
    cashAndCashEquivalents: 1484000000,
    shortTermInvestments: 91000000,
    cashAndShortTermInvestments: 1575000000,
    netReceivables: 1172000000,
    inventory: 830000000,
    otherCurrentAssets: 222000000,
    totalCurrentAssets: 3799000000,
    propertyPlantEquipmentNet: 945000000,
    goodwill: 4956000000,
    intangibleAssets: 981000000,
    goodwillAndIntangibleAssets: 4956000000,
    longTermInvestments: 185000000,
    taxAssets: 0,
    otherNonCurrentAssets: 820000000,
    totalNonCurrentAssets: 6906000000,
    otherAssets: 0,
    totalAssets: 10705000000,
    accountPayables: 446000000,
    shortTermDebt: 0,
    taxPayables: 0,
    deferredRevenue: 441000000,
    otherCurrentLiabilities: 821000000,
    totalCurrentLiabilities: 1708000000,
    longTermDebt: 2729000000,
    deferredRevenueNonCurrent: 0,
    deferredTaxLiabilitiesNonCurrent: 0,
    otherNonCurrentLiabilities: 879000000,
    totalNonCurrentLiabilities: 3608000000,
    otherLiabilities: 0,
    capitalLeaseObligations: 0,
    totalLiabilities: 5316000000,
    preferredStock: 0,
    commonStock: 3000000,
    retainedEarnings: 348000000,
    accumulatedOtherComprehensiveIncomeLoss: -282000000,
    othertotalStockholdersEquity: 5320000000,
    totalStockholdersEquity: 5389000000,
    totalEquity: 5389000000,
    totalLiabilitiesAndStockholdersEquity: 10705000000,
    minorityInterest: 0,
    totalLiabilitiesAndTotalEquity: 10705000000,
    totalInvestments: 276000000,
    totalDebt: 2729000000,
    netDebt: 1245000000,
    link: 'https://www.sec.gov/Archives/edgar/data/1090872/000109087221000027/0001090872-21-000027-index.htm',
    finalLink: 'https://www.sec.gov/Archives/edgar/data/1090872/000109087221000027/a-20211031.htm',
  },
  {
    date: '2020-10-31',
    symbol: 'A',
    reportedCurrency: 'USD',
    cik: '0001090872',
    fillingDate: '2020-12-18',
    acceptedDate: '2020-12-17 19:56:21',
    calendarYear: '2020',
    period: 'FY',
    cashAndCashEquivalents: 1441000000,
    shortTermInvestments: 0,
    cashAndShortTermInvestments: 1441000000,
    netReceivables: 1038000000,
    inventory: 720000000,
    otherCurrentAssets: 216000000,
    totalCurrentAssets: 3415000000,
    propertyPlantEquipmentNet: 845000000,
    goodwill: 4433000000,
    intangibleAssets: 831000000,
    goodwillAndIntangibleAssets: 4433000000,
    longTermInvestments: 158000000,
    taxAssets: 0,
    otherNonCurrentAssets: 776000000,
    totalNonCurrentAssets: 6212000000,
    otherAssets: 0,
    totalAssets: 9627000000,
    accountPayables: 354000000,
    shortTermDebt: 75000000,
    taxPayables: 0,
    deferredRevenue: 386000000,
    otherCurrentLiabilities: 1304000000,
    totalCurrentLiabilities: 1467000000,
    longTermDebt: 2284000000,
    deferredRevenueNonCurrent: 0,
    deferredTaxLiabilitiesNonCurrent: 0,
    otherNonCurrentLiabilities: 1003000000,
    totalNonCurrentLiabilities: 3287000000,
    otherLiabilities: 0,
    capitalLeaseObligations: 0,
    totalLiabilities: 4754000000,
    preferredStock: 0,
    commonStock: 3000000,
    retainedEarnings: 81000000,
    accumulatedOtherComprehensiveIncomeLoss: -522000000,
    othertotalStockholdersEquity: 5311000000,
    totalStockholdersEquity: 4873000000,
    totalEquity: 4873000000,
    totalLiabilitiesAndStockholdersEquity: 9627000000,
    minorityInterest: 0,
    totalLiabilitiesAndTotalEquity: 9627000000,
    totalInvestments: 158000000,
    totalDebt: 2359000000,
    netDebt: 918000000,
    link: 'https://www.sec.gov/Archives/edgar/data/1090872/000109087220000020/0001090872-20-000020-index.htm',
    finalLink: 'https://www.sec.gov/Archives/edgar/data/1090872/000109087220000020/a-20201031.htm',
  },
  {
    date: '2019-10-31',
    symbol: 'A',
    reportedCurrency: 'USD',
    cik: '0001090872',
    fillingDate: '2019-12-19',
    acceptedDate: '2019-12-19 17:11:01',
    calendarYear: '2019',
    period: 'FY',
    cashAndCashEquivalents: 1382000000,
    shortTermInvestments: 0,
    cashAndShortTermInvestments: 1382000000,
    netReceivables: 930000000,
    inventory: 679000000,
    otherCurrentAssets: 198000000,
    totalCurrentAssets: 3189000000,
    propertyPlantEquipmentNet: 850000000,
    goodwill: 4700000000,
    intangibleAssets: 1107000000,
    goodwillAndIntangibleAssets: 4700000000,
    longTermInvestments: 102000000,
    taxAssets: 0,
    otherNonCurrentAssets: 611000000,
    totalNonCurrentAssets: 6263000000,
    otherAssets: 0,
    totalAssets: 9452000000,
    accountPayables: 354000000,
    shortTermDebt: 616000000,
    taxPayables: 0,
    deferredRevenue: 336000000,
    otherCurrentLiabilities: 1548000000,
    totalCurrentLiabilities: 2080000000,
    longTermDebt: 1791000000,
    deferredRevenueNonCurrent: 0,
    deferredTaxLiabilitiesNonCurrent: 0,
    otherNonCurrentLiabilities: 833000000,
    totalNonCurrentLiabilities: 2624000000,
    otherLiabilities: 0,
    capitalLeaseObligations: 0,
    totalLiabilities: 4704000000,
    preferredStock: 0,
    commonStock: 3000000,
    retainedEarnings: -18000000,
    accumulatedOtherComprehensiveIncomeLoss: -514000000,
    othertotalStockholdersEquity: 5277000000,
    totalStockholdersEquity: 4748000000,
    totalEquity: 4748000000,
    totalLiabilitiesAndStockholdersEquity: 9452000000,
    minorityInterest: 0,
    totalLiabilitiesAndTotalEquity: 9452000000,
    totalInvestments: 102000000,
    totalDebt: 2407000000,
    netDebt: 1025000000,
    link: 'https://www.sec.gov/Archives/edgar/data/1090872/000109087219000022/0001090872-19-000022-index.htm',
    finalLink:
      'https://www.sec.gov/Archives/edgar/data/1090872/000109087219000022/a-10312019x10k.htm',
  },
  {
    date: '2018-10-31',
    symbol: 'A',
    reportedCurrency: 'USD',
    cik: '0001090872',
    fillingDate: '2018-12-20',
    acceptedDate: '2018-12-20 16:27:07',
    calendarYear: '2018',
    period: 'FY',
    cashAndCashEquivalents: 2247000000,
    shortTermInvestments: 0,
    cashAndShortTermInvestments: 2247000000,
    netReceivables: 776000000,
    inventory: 638000000,
    otherCurrentAssets: 187000000,
    totalCurrentAssets: 3848000000,
    propertyPlantEquipmentNet: 822000000,
    goodwill: 2973000000,
    intangibleAssets: 491000000,
    goodwillAndIntangibleAssets: 3464000000,
    longTermInvestments: 68000000,
    taxAssets: 0,
    otherNonCurrentAssets: 339000000,
    totalNonCurrentAssets: 4693000000,
    otherAssets: 0,
    totalAssets: 8541000000,
    accountPayables: 340000000,
    shortTermDebt: 0,
    taxPayables: 0,
    deferredRevenue: 324000000,
    otherCurrentLiabilities: 507000000,
    totalCurrentLiabilities: 1171000000,
    longTermDebt: 1799000000,
    deferredRevenueNonCurrent: 0,
    deferredTaxLiabilitiesNonCurrent: 0,
    otherNonCurrentLiabilities: 2135000000,
    totalNonCurrentLiabilities: 2799000000,
    otherLiabilities: 0,
    capitalLeaseObligations: 0,
    totalLiabilities: 3970000000,
    preferredStock: 0,
    commonStock: 3000000,
    retainedEarnings: -336000000,
    accumulatedOtherComprehensiveIncomeLoss: -408000000,
    othertotalStockholdersEquity: 5308000000,
    totalStockholdersEquity: 4567000000,
    totalEquity: 4567000000,
    totalLiabilitiesAndStockholdersEquity: 8541000000,
    minorityInterest: 4000000,
    totalLiabilitiesAndTotalEquity: 8541000000,
    totalInvestments: 68000000,
    totalDebt: 1799000000,
    netDebt: -448000000,
    link: 'https://www.sec.gov/Archives/edgar/data/1090872/000109087218000019/0001090872-18-000019-index.htm',
    finalLink:
      'https://www.sec.gov/Archives/edgar/data/1090872/000109087218000019/a-10312018x10k.htm',
  },
];

export default mockBalanceSheetStatement;
