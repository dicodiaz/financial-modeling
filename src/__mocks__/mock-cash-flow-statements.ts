import { CashFlowStatementType } from '../redux/utils/types';

const mockCashFlowStatements: CashFlowStatementType[] = [
  {
    date: '2022-10-31',
    symbol: 'A',
    reportedCurrency: 'USD',
    cik: '0001090872',
    fillingDate: '2022-12-21',
    acceptedDate: '2022-12-20 18:42:30',
    calendarYear: '2022',
    period: 'FY',
    netIncome: 1254000000,
    depreciationAndAmortization: 317000000,
    deferredIncomeTax: 8000000,
    stockBasedCompensation: 125000000,
    changeInWorkingCapital: -478000000,
    accountsReceivables: -321000000,
    inventory: -248000000,
    accountsPayables: 121000000,
    otherWorkingCapital: -30000000,
    otherNonCashItems: 86000000,
    netCashProvidedByOperatingActivities: 1312000000,
    investmentsInPropertyPlantAndEquipment: -291000000,
    acquisitionsNet: -52000000,
    purchasesOfInvestments: -13000000,
    salesMaturitiesOfInvestments: 22000000,
    otherInvestingActivites: -4000000,
    netCashUsedForInvestingActivites: -338000000,
    debtRepayment: -1869000000,
    commonStockIssued: 58000000,
    commonStockRepurchased: -1139000000,
    dividendsPaid: -250000000,
    otherFinancingActivites: 1828000000,
    netCashUsedProvidedByFinancingActivities: -1372000000,
    effectOfForexChangesOnCash: -36000000,
    netChangeInCash: -434000000,
    cashAtEndOfPeriod: 1056000000,
    cashAtBeginningOfPeriod: 1490000000,
    operatingCashFlow: 1312000000,
    capitalExpenditure: -291000000,
    freeCashFlow: 1021000000,
    link: 'https://www.sec.gov/Archives/edgar/data/1090872/000109087222000026/0001090872-22-000026-index.htm',
    finalLink: 'https://www.sec.gov/Archives/edgar/data/1090872/000109087222000026/a-20221031.htm',
  },
  {
    date: '2021-10-31',
    symbol: 'A',
    reportedCurrency: 'USD',
    cik: '0001090872',
    fillingDate: '2021-12-17',
    acceptedDate: '2021-12-17 17:01:49',
    calendarYear: '2021',
    period: 'FY',
    netIncome: 1210000000,
    depreciationAndAmortization: 321000000,
    deferredIncomeTax: 14000000,
    stockBasedCompensation: 110000000,
    changeInWorkingCapital: -88000000,
    accountsReceivables: -128000000,
    inventory: -136000000,
    accountsPayables: 64000000,
    otherWorkingCapital: 112000000,
    otherNonCashItems: -82000000,
    netCashProvidedByOperatingActivities: 1485000000,
    investmentsInPropertyPlantAndEquipment: -189000000,
    acquisitionsNet: -546000000,
    purchasesOfInvestments: -22000000,
    salesMaturitiesOfInvestments: 12000000,
    otherInvestingActivites: -5000000,
    netCashUsedForInvestingActivites: -749000000,
    debtRepayment: -2139000000,
    commonStockIssued: 2550000000,
    commonStockRepurchased: -788000000,
    dividendsPaid: -236000000,
    otherFinancingActivites: 2467000000,
    netCashUsedProvidedByFinancingActivities: -696000000,
    effectOfForexChangesOnCash: 3000000,
    netChangeInCash: 43000000,
    cashAtEndOfPeriod: 1490000000,
    cashAtBeginningOfPeriod: 1447000000,
    operatingCashFlow: 1485000000,
    capitalExpenditure: -189000000,
    freeCashFlow: 1296000000,
    link: 'https://www.sec.gov/Archives/edgar/data/1090872/000109087221000027/0001090872-21-000027-index.htm',
    finalLink: 'https://www.sec.gov/Archives/edgar/data/1090872/000109087221000027/a-20211031.htm',
  },
  {
    date: '2020-10-31',
    symbol: 'A',
    reportedCurrency: 'USD',
    cik: '0001090872',
    fillingDate: '2020-12-18',
    acceptedDate: '2020-12-17 19:56:21',
    calendarYear: '2020',
    period: 'FY',
    netIncome: 719000000,
    depreciationAndAmortization: 308000000,
    deferredIncomeTax: 29000000,
    stockBasedCompensation: 83000000,
    changeInWorkingCapital: -296000000,
    accountsReceivables: -107000000,
    inventory: -68000000,
    accountsPayables: 2000000,
    otherWorkingCapital: -123000000,
    otherNonCashItems: 78000000,
    netCashProvidedByOperatingActivities: 921000000,
    investmentsInPropertyPlantAndEquipment: -119000000,
    acquisitionsNet: 0,
    purchasesOfInvestments: -20000000,
    salesMaturitiesOfInvestments: 0,
    otherInvestingActivites: -8000000,
    netCashUsedForInvestingActivites: -147000000,
    debtRepayment: -1762000000,
    commonStockIssued: 1777000000,
    commonStockRepurchased: -469000000,
    dividendsPaid: -222000000,
    otherFinancingActivites: 1736000000,
    netCashUsedProvidedByFinancingActivities: -717000000,
    effectOfForexChangesOnCash: 2000000,
    netChangeInCash: 59000000,
    cashAtEndOfPeriod: 1447000000,
    cashAtBeginningOfPeriod: 1388000000,
    operatingCashFlow: 921000000,
    capitalExpenditure: -119000000,
    freeCashFlow: 802000000,
    link: 'https://www.sec.gov/Archives/edgar/data/1090872/000109087220000020/0001090872-20-000020-index.htm',
    finalLink: 'https://www.sec.gov/Archives/edgar/data/1090872/000109087220000020/a-20201031.htm',
  },
  {
    date: '2019-10-31',
    symbol: 'A',
    reportedCurrency: 'USD',
    cik: '0001090872',
    fillingDate: '2019-12-19',
    acceptedDate: '2019-12-19 17:11:01',
    calendarYear: '2019',
    period: 'FY',
    netIncome: 1071000000,
    depreciationAndAmortization: 238000000,
    deferredIncomeTax: -255000000,
    stockBasedCompensation: 72000000,
    changeInWorkingCapital: -388000000,
    accountsReceivables: -106000000,
    inventory: -36000000,
    accountsPayables: 30000000,
    otherWorkingCapital: -276000000,
    otherNonCashItems: 283000000,
    netCashProvidedByOperatingActivities: 1021000000,
    investmentsInPropertyPlantAndEquipment: -156000000,
    acquisitionsNet: -1408000000,
    purchasesOfInvestments: -23000000,
    salesMaturitiesOfInvestments: -1000000,
    otherInvestingActivites: -4000000,
    netCashUsedForInvestingActivites: -1590000000,
    debtRepayment: -512000000,
    commonStockIssued: 1166000000,
    commonStockRepurchased: -723000000,
    dividendsPaid: -206000000,
    otherFinancingActivites: 1142000000,
    netCashUsedProvidedByFinancingActivities: -299000000,
    effectOfForexChangesOnCash: 2000000,
    netChangeInCash: -866000000,
    cashAtEndOfPeriod: 1388000000,
    cashAtBeginningOfPeriod: 2254000000,
    operatingCashFlow: 1021000000,
    capitalExpenditure: -156000000,
    freeCashFlow: 865000000,
    link: 'https://www.sec.gov/Archives/edgar/data/1090872/000109087219000022/0001090872-19-000022-index.htm',
    finalLink:
      'https://www.sec.gov/Archives/edgar/data/1090872/000109087219000022/a-10312019x10k.htm',
  },
  {
    date: '2018-10-31',
    symbol: 'A',
    reportedCurrency: 'USD',
    cik: '0001090872',
    fillingDate: '2018-12-20',
    acceptedDate: '2018-12-20 16:27:07',
    calendarYear: '2018',
    period: 'FY',
    netIncome: 316000000,
    depreciationAndAmortization: 210000000,
    deferredIncomeTax: -16000000,
    stockBasedCompensation: 70000000,
    changeInWorkingCapital: 455000000,
    accountsReceivables: -65000000,
    inventory: -83000000,
    accountsPayables: 40000000,
    otherWorkingCapital: 563000000,
    otherNonCashItems: 52000000,
    netCashProvidedByOperatingActivities: 1087000000,
    investmentsInPropertyPlantAndEquipment: -177000000,
    acquisitionsNet: -516000000,
    purchasesOfInvestments: -11000000,
    salesMaturitiesOfInvestments: 0,
    otherInvestingActivites: 0,
    netCashUsedForInvestingActivites: -704000000,
    debtRepayment: -693000000,
    commonStockIssued: 0,
    commonStockRepurchased: -422000000,
    dividendsPaid: -191000000,
    otherFinancingActivites: 509000000,
    netCashUsedProvidedByFinancingActivities: -797000000,
    effectOfForexChangesOnCash: -17000000,
    netChangeInCash: -431000000,
    cashAtEndOfPeriod: 2247000000,
    cashAtBeginningOfPeriod: 2678000000,
    operatingCashFlow: 1087000000,
    capitalExpenditure: -177000000,
    freeCashFlow: 910000000,
    link: 'https://www.sec.gov/Archives/edgar/data/1090872/000109087218000019/0001090872-18-000019-index.htm',
    finalLink:
      'https://www.sec.gov/Archives/edgar/data/1090872/000109087218000019/a-10312018x10k.htm',
  },
];

export default mockCashFlowStatements;
