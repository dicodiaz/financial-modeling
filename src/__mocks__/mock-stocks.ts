import { StockType, STOCK_EXCHANGES, STOCK_TYPES } from '../redux/utils/types';

const mockStocks: StockType[] = [
  {
    symbol: 'A',
    name: 'Agilent Technologies, Inc.',
    price: 117,
    exchange: 'New York Stock Exchange',
    exchangeShortName: STOCK_EXCHANGES.NYSE,
    type: STOCK_TYPES.Stock,
  },
  {
    symbol: 'AA',
    name: 'Alcoa Corporation',
    price: 34.09,
    exchange: 'New York Stock Exchange',
    exchangeShortName: STOCK_EXCHANGES.NYSE,
    type: STOCK_TYPES.Stock,
  },
  {
    symbol: 'AAA',
    name: 'AXS First Priority CLO Bond ETF',
    price: 24.58,
    exchange: 'New York Stock Exchange Arca',
    exchangeShortName: STOCK_EXCHANGES.AMEX,
    type: STOCK_TYPES.ETF,
  },
  {
    symbol: 'AABPX',
    name: 'American Beacon Balanced Fund',
    price: 11.09,
    exchange: 'Nasdaq',
    exchangeShortName: STOCK_EXCHANGES.NASDAQ,
    type: STOCK_TYPES.Trust,
  },
  {
    symbol: 'AAC',
    name: 'Ares Acquisition Corporation',
    price: 10.57,
    exchange: 'New York Stock Exchange',
    exchangeShortName: STOCK_EXCHANGES.NYSE,
    type: STOCK_TYPES.Stock,
  },
  {
    symbol: 'AAC-UN',
    name: 'Ares Acquisition Corporation',
    price: 10.7101,
    exchange: 'New York Stock Exchange',
    exchangeShortName: STOCK_EXCHANGES.NYSE,
    type: STOCK_TYPES.Stock,
  },
  {
    symbol: 'AAC-WT',
    name: 'Ares Acquisition Corporation Re',
    price: 0.85,
    exchange: 'New York Stock Exchange',
    exchangeShortName: STOCK_EXCHANGES.NYSE,
    type: STOCK_TYPES.Stock,
  },
  {
    symbol: 'AACG',
    name: 'ATA Creativity Global',
    price: 1.38,
    exchange: 'NASDAQ Global Market',
    exchangeShortName: STOCK_EXCHANGES.NASDAQ,
    type: STOCK_TYPES.Stock,
  },
  {
    symbol: 'AACI',
    name: 'Armada Acquisition Corp. I',
    price: 10.41,
    exchange: 'NASDAQ Global Market',
    exchangeShortName: STOCK_EXCHANGES.NASDAQ,
    type: STOCK_TYPES.Stock,
  },
  {
    symbol: 'AACIU',
    name: 'Armada Acquisition Corp. I',
    price: 10.3,
    exchange: 'NASDAQ Global Market',
    exchangeShortName: STOCK_EXCHANGES.NASDAQ,
    type: STOCK_TYPES.Stock,
  },
];

export default mockStocks;
