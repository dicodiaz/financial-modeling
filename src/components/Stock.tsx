import { FC } from 'react';
import { Card } from 'react-bootstrap';
import { useNavigate } from 'react-router-dom';
import { StockType } from '../redux/utils/types';
import StatementItem from './StatementItem';

export type StockProps = {
  data: StockType;
};

const Stock: FC<StockProps> = ({ data }) => {
  const navigate = useNavigate();
  const { symbol } = data;

  return (
    <Card role="button" className="rounded-0 bg-stock text-white" onClick={() => navigate(symbol)}>
      <Card.Body>
        <Card.Title>{symbol}</Card.Title>
        {Object.entries(data).map(([name, value]) => {
          if (['symbol', 'exchange'].includes(name)) {
            return null;
          }

          return <StatementItem key={name} name={name} value={value} reportedCurrency="USD" />;
        })}
      </Card.Body>
    </Card>
  );
};

export default Stock;
