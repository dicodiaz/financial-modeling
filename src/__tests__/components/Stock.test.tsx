import Stock from '../../components/Stock';
import { mockStock } from '../../__mocks__/mock-stocks';
import { renderWithProviders } from '../test-utils';

describe('Stock', () => {
  it('should render a clickable card with the expected content', () => {
    const { container } = renderWithProviders(<Stock data={mockStock} />);
    expect(container).toMatchSnapshot();
  });
});
