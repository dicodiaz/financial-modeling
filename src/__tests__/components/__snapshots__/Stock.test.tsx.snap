// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Stock > should render a clickable card with the expected content 1`] = `
<div>
  <div
    class="rounded-0 bg-stock text-white card"
    role="button"
  >
    <div
      class="card-body"
    >
      <div
        class="card-title h5"
      >
        A
      </div>
      <div>
        <p
          class="mb-0 fw-bold"
        >
          Name
        </p>
        <p
          class="mb-0 ps-2 small fst-italic"
        >
          Agilent Technologies, Inc.
        </p>
      </div>
      <div>
        <p
          class="mb-0 fw-bold"
        >
          Price
        </p>
        <p
          class="mb-0 ps-2 small fst-italic"
        >
          US$Â 117
        </p>
      </div>
      <div>
        <p
          class="mb-0 fw-bold"
        >
          Exchange
        </p>
        <p
          class="mb-0 ps-2 small fst-italic"
        >
          NYSE
        </p>
      </div>
      <div>
        <p
          class="mb-0 fw-bold"
        >
          Type
        </p>
        <p
          class="mb-0 ps-2 small fst-italic"
        >
          stock
        </p>
      </div>
    </div>
  </div>
</div>
`;
